daft/catalog/__init__.py

```diff
@@ -72,7 +72,7 @@
 def unregister_catalog(catalog_name: str | None) -> bool:
     """Unregisters a catalog from the Daft catalog system.
 
-    DEPRECATED: This is deprecated and will be removed in daft >= 0.5.0; please use `daft.detach_catalog`.
+    !!! warning "DEPRECATED: This is deprecated and will be removed in daft >= 0.5.0; please use [daft.detach_catalog][daft.Session.detach_catalog]."
```

Comment:
these `[daft.detach_catalog][daft.Session.detach_catalog]` don't resolve properly in any IDE. 

<img width="692" alt="image" src="https://github.com/user-attachments/assets/e53c5c73-090d-41f8-b74a-701f9c5f82ab" />

<img width="331" alt="image" src="https://github.com/user-attachments/assets/c2ec5d3e-6434-4583-818d-ac4d051e7893" />